central/resourcecollection/service/service_impl.go

```diff
@@ -83,3 +94,58 @@ func (s *serviceImpl) getCollection(ctx context.Context, id string) (*v1.GetColl
 		Deployments: nil,
 	}, nil
 }
+
+// CreateCollection creates a new collection from the given request
+func (s *serviceImpl) CreateCollection(ctx context.Context, request *v1.CreateCollectionRequest) (*v1.CreateCollectionResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, nil
+	}
+
+	if request.GetName() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Collection name should not be empty")
+	}
+
+	collection, err := collectionRequestToCollection(ctx, request, true)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.datastore.AddCollection(ctx, collection)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.CreateCollectionResponse{Collection: collection}, nil
+}
+
+func collectionRequestToCollection(ctx context.Context, request collectionRequest, isCreate bool) (*storage.ResourceCollection, error) {
+	user := utils.UserFromContext(ctx)
+	if user == nil {
+		return nil, errors.New("User identity not provided")
```

Comment:
because it implied from the context `could not determine user identity` seems more appropriate